functionary/builder/models/build.py

```diff
@@ -56,6 +55,29 @@ class Meta:
     def __str__(self):
         return f"{self.id}"
 
+    def error(self):
+        """Saves status as `ERROR`"""
+        self._update_status(self.ERROR)
+
+    def pending(self):
+        """Saves status as `PENDING`"""
+        self._update_status(self.PENDING)
+
+    def complete(self):
+        """Saves status as `COMPLETE`"""
+        self._update_status(self.COMPLETE)
+        self.package.complete()
+
+    def in_progress(self):
+        """Saves status as `IN_PROGRESS`"""
+        self._update_status(self.IN_PROGRESS)
+
+    def _update_status(self, status: str) -> None:
+        """Update status of the build"""
+        if status != self.status:
+            self.status = status
+            self.save(update_fields=["status"])
```

Comment:
I commented elsewhere  on this, but please don't do this.  I have a hard time thinking of a situation where the benefit of doing this outweighs the pain it will likely cause someone one day.